src/zcash/Note.cpp

```diff
@@ -0,0 +1,92 @@
+#include "Note.hpp"
+#include "prf.h"
+#include "crypto/sha256.h"
+#include "zerocash/utils/util.h"
+
+#include "version.h"
+#include "streams.h"
+
+namespace libzcash {
+
+Note::Note() {
+    a_pk = random_uint256();
+    rho = random_uint256();
+    r = random_uint256();
+    value = 0;
+}
+
+uint256 Note::cm() const {
+    unsigned char discriminant = 0xb0;
+
+    CSHA256 hasher;
+    hasher.Write(&discriminant, 1);
+    hasher.Write(a_pk.begin(), 32);
+
+    std::vector<unsigned char> value_vec(sizeof(value), 0);
+    libzerocash::convertIntToBytesVector(value, value_vec);
```

Comment:
This has the wrong endianness. The spec says "All integers in Zcash-specific encodings are unsigned, have fixed bit length, and are encoded in little-endian byte order." But `convertIntToBytesVector's code is...

```
void convertIntToBytesVector(const uint64_t val_int, std::vector<unsigned char>& bytes) {
     for(size_t i = 0; i < bytes.size(); i++) {
         bytes[bytes.size()-1-i] = (val_int >> (i * 8));
    }
}
```

...which is big endian.